src/Microsoft.IdentityModel.Tokens/Validators.cs

```diff
@@ -213,16 +213,29 @@ public static string ValidateIssuer(string issuer, SecurityToken securityToken,
         /// <returns>The issuer to use when creating the "Claim"(s) in a "ClaimsIdentity".</returns>
         /// <exception cref="ArgumentNullException">If 'validationParameters' is null.</exception>
         /// <exception cref="ArgumentNullException">If 'issuer' is null or whitespace and <see cref="TokenValidationParameters.ValidateIssuer"/> is true.</exception>
-        /// <exception cref="ArgumentNullException">If ' configuration' is null.</exception>
+        /// <exception cref="ArgumentNullException">If 'configuration' is null.</exception>
         /// <exception cref="SecurityTokenInvalidIssuerException">If <see cref="TokenValidationParameters.ValidIssuer"/> is null or whitespace and <see cref="TokenValidationParameters.ValidIssuers"/> is null and <see cref="BaseConfiguration.Issuer"/> is null.</exception>
         /// <exception cref="SecurityTokenInvalidIssuerException">If 'issuer' failed to matched either <see cref="TokenValidationParameters.ValidIssuer"/> or one of <see cref="TokenValidationParameters.ValidIssuers"/> or <see cref="BaseConfiguration.Issuer"/>.</exception>
         /// <remarks>An EXACT match is required.</remarks>
         internal static string ValidateIssuer(string issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
         {
-            ValueTask<string> vt = ValidateIssuerAsync(issuer, securityToken, validationParameters, configuration);
-            return vt.IsCompletedSuccessfully ?
-                vt.Result :
-                vt.AsTask().ConfigureAwait(false).GetAwaiter().GetResult();
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+            // Given that this is a synchronous method and to avoid consuming excesive threads
+            // we will favor the synchronous delegates over the asynchronous one.
+            if (validationParameters.IssuerValidator == null
```

Comment:
this is a behavioral breaking change?

Reply:
Or am I reading it wrong?

Reply:
> this is a behavioral breaking change?

👍 Currently IssuerValidatorAsync  is always preferred: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b352bcce8cf2c1676f6ef13558aff29b7766602f/src/Microsoft.IdentityModel.Tokens/Validators.cs#L248-L264

Reply:
Correct this changes introduces a change on the behavior. Async custom validators would be preferred when running asynchronously and Sync custom validators otherwise.

Reply:
We need a coordinated change across all the delegates that are async and do not rely on throwing, but instead return (too be designed) a ValidationResult.

We have to address multiple policies from SDK's that build on top. This is where the first policy fails, throws and logs a bunch of errors, and the second one succeeds.

We have thrown AND logged errors, but the authentication succeeded.
Users should have a way to capture all results, but logging and throwing by default is causing performance and debugging issues as logs have a lot of false negatives.